### ISSUE TITLE
Unload mods when shutting down GameRunner

### ISSUE THREAD

**Issue by paulcscharf** [beardgame/td#4]
Most of the resources are managed and we don't care, but we do have to clean up all the OpenGL stuff:

- [x] surfaces (includes index and vertex buffers)
- [x] textures (not actually stored anywhere right now, so this needs some more thinking)

Probably have to turn the PackedSpriteSet into an interface so that we can hide more stuff in them, like the cleanup itself perhaps. Then we could also remove the funny IHasSurface interface (in place of maybe another one that serves the same purpose but does it better.

**Comment by paulcscharf** [comment]
Or actually, we could also have the methods publicly on the sprite set content. Make the content itself aware that it can be unloaded/disposed.

**Comment by paulcscharf** [comment]
I indeed removed the interface and simply put the surface into the PackedSpriteSet. Had to add a bunch of disposing to awgraphics which probably needs to be polished in the future. I also unload materials which were added recently, and I made shaders be reused, instead of unloading them, because that's a little less annoying to do and will work just fine (and slightly speed up loading times for consecutive games).
